### PR TITLE
JS: improve the sql-injection help page

### PR DESCRIPTION
**Pull request by erik-krogh**
See the backref for motivation. 

I tried to write some more concrete examples for both SQL/NoSQL injection. 

The QHelp preview just doesn't work well with include files.  
But here is the new rendered qhelp: https://gist.github.com/erik-krogh/86c0b42fbd5371295cb9ceda4929410b

### CODE REVIEW

File: javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp [hunk 1]
```diff
@@ -32,31 +38,61 @@ an HTTP request handler in a web application, whose parameter
 </p>
 
 <p>
-The handler constructs two copies of the same SQL query involving
-user input taken from the request object, once unsafely using
-string concatenation, and once safely using query parameters.
+The handler constructs constructs an SQL query string from user input
+and executes it as a database query using the <code>pg</code> library. 
+THe user input may contain quote characters, so this code is vulnerable
```
Comment (asgerf) [inline javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp:43]:
```suggestion
The user input may contain quote characters, so this code is vulnerable
```

File: javascript/ql/src/Security/CWE-089/examples/SqlInjectionFix.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const app = require("express")(),
+      pg = require("pg"),
+      pool = new pg.Pool(config);
+
+app.get("search", function handler(req, res) {
+  // GOOD: use parameters
+  var query2 =
+    "SELECT ITEM,PRICE FROM PRODUCT WHERE ITEM_CATEGORY=$1" + " ORDER BY PRICE";
```
Comment (asgerf) [inline javascript/ql/src/Security/CWE-089/examples/SqlInjectionFix.js:8]: It's a bit strange to have the concatenation operator here, though I can see it makes it more obvious that `$1` is a literal replacement for the `'" + xxx + "'` part.

If you'd rather keep the operator, could we perhaps move the `ORDER BY PRICE` to the following line?

**Review by asgerf — COMMENTED** [review]
👍

**Comment by erik-krogh** [comment]
Lets get a 👍 from the docs team.

**Review by asgerf — APPROVED** [review]
👍 from a technical perspective

**Comment by mchammer01** [comment]
I'll review this on behalf of docs!

File: javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp [hunk 3]
```diff
@@ -32,31 +38,61 @@ an HTTP request handler in a web application, whose parameter
 </p>
 
 <p>
-The handler constructs two copies of the same SQL query involving
-user input taken from the request object, once unsafely using
-string concatenation, and once safely using query parameters.
+The handler constructs constructs an SQL query string from user input
```
Comment (mchammer01) [inline javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp:41]: Remove the repeated word?
```suggestion
The handler constructs an SQL query string from user input
```

File: javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp [hunk 4]
```diff
@@ -32,31 +38,61 @@ an HTTP request handler in a web application, whose parameter
...
+<p>
+Alternatively, we can use a library like <code>sqlstring</code> to
+escape the user input before embedding it into the query string:
+</p>
+<sample src="examples/SqlInjectionFix2.js" />
+</example>
+
+<example>
+<p>
+In the following example an express handler attempts to delete
```
Comment (mchammer01) [inline javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp:68]: Nit
```suggestion
In the following example, an express handler attempts to delete
```

File: javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp [hunk 5]
```diff
@@ -32,31 +38,61 @@ an HTTP request handler in a web application, whose parameter
...
+a single document from a MongoDB collection. The document to be
+deleted is identified by its <code>_id</code> field, which is
+constructed from user input. The user input may contain a query
+object, so this code is vulnerable to a NoSQL injection attack.
+</p>
+
+<sample src="examples/NoSqlInjection.js" />
+
+<p>
+To fix this vulnerability we can use the <code>$eq</code> operator
```
Comment (mchammer01) [inline javascript/ql/src/Security/CWE-089/SqlInjection.inc.qhelp:78]: Nit
```suggestion
To fix this vulnerability, we can use the <code>$eq</code> operator
```

**Review by mchammer01 — APPROVED** [review]
@erik-krogh 👋🏻 - this LGTM from an editorial point of view.
I've commented on a few minor things.

**Review by mchammer01 — APPROVED** [review]
LGTM :shipit:

**Pull request merged**